src/analyses/goto_check.cpp

```diff
@@ -1438,13 +1438,16 @@ void goto_checkt::add_guarded_claim(
     goto_program_instruction_typet type=
       enable_assert_to_assume?ASSUME:ASSERT;
 
-    goto_programt::targett t=new_code.add_instruction(type);
+    goto_programt::targett t = new_code.add(goto_programt::instructiont(
+      static_cast<const codet &>(get_nil_irep()),
+      source_location,
+      type,
+      std::move(new_expr),
+      {}));
```

Comment:
How about using an `if`/`else` branching here to avoid the use of this very generic constructor? I think the amount of code to be duplicated is very modest.